### PR TITLE
chore(core): delete dialog with tests

### PR DESCRIPTION
**Pull request by barsukov**
# Summary

Added comprehensive test coverage for the `DeleteClusterDialog` component using Vitest and React Testing Library. This PR introduces a complete test suite that validates component rendering, user interactions, and callback behaviors while treating the component as a blackbox without mocking external dependencies.

**Note: This implementation needs i18n support for proper internationalization of dialog text and messages.**

# Changes Made

- Added comprehensive test suite for `DeleteClusterDialog` component
- Implemented blackbox testing approach without mocking Juno UI components
- Added tests for modal visibility states (open/closed)
- Added tests for cluster name display and dynamic content rendering
- Added tests for user interactions (delete, cancel, keyboard navigation)
- Added tests for accessibility features and button roles
- Added tests for edge cases (special characters, long names, empty values)
- Added tests for component prop changes and state management
- Configured proper test setup with PortalProvider wrapper
- Added comprehensive assertions for warning messages and dialog content

# Related Issues

- Feature Request: Add test coverage for DeleteClusterDialog component
- **TODO**: Add i18n support for dialog text internationalization

# Screenshots (if applicable)

N/A - This PR only adds test coverage without visual changes.

# Testing Instructions

1. `pnpm i`
2. `pnpm run test DeleteClusterDialog` - Run specific component tests
3. `pnpm run test` - Run all tests to ensure no regressions
4. `pnpm run test:coverage` - Verify test coverage metrics
5. Verify all test cases pass:
   - Modal rendering and visibility
   - Content display with different cluster names
   - User interaction callbacks (delete/cancel)
   - Accessibility and keyboard navigation
   - Edge cases and error handling

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.
- [ ] **TODO**: Add i18n support for internationalization of dialog text and messages.

### CODE REVIEW

**Review by andypf — DISMISSED** [review]

File: apps/aurora-portal/src/client/main.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ i18n.load({
   en: messages,
   de: messagesDe,
 })
-i18n.activate("en")
+i18n.activate("de")
```
Comment (andypf) [inline apps/aurora-portal/src/client/main.tsx:12]: Does it mean German is default?

File: apps/aurora-portal/src/client/routes/gardener/-components/DeleteClusterDialog.tsx [hunk 2]
```diff
@@ -15,79 +15,71 @@ export const DeleteClusterDialog: React.FC<DeleteClusterGardenerDialogProps> = (
...
+                  <Trans>This action cannot be undone. The cluster will be permanently deleted.</Trans>
+                </Message>
 
-              {/* Consequence */}
-              <p className="text-aurora-gray-300 text-base text-left">
-                After continuing, your project will no longer have access to the{" "}
-                <strong className="text-aurora-white font-semibold">{clusterName}</strong> resources.
-              </p>
+                {/* Question */}
+                <p className="text-juno-grey-light text-base text-left">
```
[the remark this answers is not in the document]
Comment (TilmanHaupt) [reply apps/aurora-portal/src/client/routes/gardener/-components/DeleteClusterDialog.tsx:64]: > text-theme-light

[the remark this answers is not in the document]
Comment (barsukov) [reply apps/aurora-portal/src/client/routes/gardener/-components/DeleteClusterDialog.tsx:64]: still did not get it a bit
